### ISSUE TITLE
Dead PFTs in PPE2_BHSon simulations

### ISSUE THREAD

**Issue by wwieder** [ESCOMP/CTSM#1274]
## Issue:
Decid. Temp trees and C4 grasses are dead at the end of the historical period in [`PPE2_BHSon` simulations](https://www.cgd.ucar.edu/tss/clm/diagnostics/clm5.1_dev/pft_level_plots/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_hist_pft_2dPDF_whittaker-1.png).
  - [ ] **Confirm dead PFTs are occurring at the end of pAD in the `PPE2_BHSon` case**  @olyson 

## Notes
At the end of AD spinup decid. temperate trees and C4 grasses look OK in [`LMBIRCH`](https://www.cgd.ucar.edu/tss/clm/diagnostics/clm5.1_dev/pft_level_plots/clm51_lmbirchwkattgen01ctsm51d006_2deg_GSWP3V1_nocrop_PPE_1850AD_pft_2dPDF_whittaker-1.png), PPE2_BHSoff and [`PPE2_BHSon`](https://www.cgd.ucar.edu/tss/clm/diagnostics/clm5.1_dev/pft_level_plots/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_1850AD_pft_2dPDF_whittaker-1.png) simulations.
  - [ ] **Continue the `PPE2_BHSoff` case through pAD to see if PFTs survive** @olyson 

## Ideas:
- Assuming this can be isolated to `BHSon` simulations, are the dead PFTs reflecting N limitation caused by [cooler soil temperatures ](https://webext.cgd.ucar.edu/I20TR/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_hist/lnd/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_hist.1995_2014-clm51_lmbirchwkattgen01ctsm51d006_2deg_GSWP3V1_nocrop_PPE2_hist.1995_2014/set2/set2_ANN_TSOI_4.png)with `BHSon` that would slow N mineralization rates?
- The N sensitivity of decid. and C4 grass pft's is consistent with behavior of the model during CLM5 development & would not be immediately evident during AD spinup when decomposition and N mineralization rates are higher

### Ideas to try in parallel to explore this and other possibilities:
- [ ] Regional simulations in NE U.S. w/ Keith's code to replicate features of `BHSoff` & `BHSon`, @swensosc 
- [ ] Fertilizing `BHSon` simulation in pAD by pointing to: stream_fldfilename_ndep = '/glade/work/wwieder/modified_inputs/fndep_clm_WACCM6_CMIP6piControl001_y21-50avg_1850monthly_0.95x1.25_c180802_fert05.nc' , @swensosc 
- [ ] @oleson is it worth branching your global BHSon-pAD simulation with the extra nitrogen deposition? 
- [ ] Look into parameter changes related to `ncosts` that may alleviate this issue, these can be tested in Sean's regional simulations? @wwieder @dlawrenncar

**Comment by olyson** [comment]
BL Deciduous Temperate Trees are still at 84% survivability at the end of the pAD.  
C4 grass is at 74%.
I'll have some spatial plots once timeseries generation is done.

**Comment by olyson** [comment]
I've generated some survivability timeseries plots for the last 200 years of the pAD:

https://www.cgd.ucar.edu/tss/clm/diagnostics/clm5.1_dev/pft_level_plots/plot_checkfordead_timeseries_PPEFINAL_1850pAD_TLAI.png

and the historical:

https://www.cgd.ucar.edu/tss/clm/diagnostics/clm5.1_dev/pft_level_plots/plot_checkfordead_timeseries_PPEFINAL_hist_TLAI.png

The BL deciduous temperate and boreal are stable in the pAD, but then die off in the historical.

The C4 grass declines only slightly in the historical but I think the damage was done earlier in the AD/pAD.

**Comment by dlawrenncar** [comment]
Interesting.  Looking through diagnostics plots, it looks to me like the
vegetation in eastern US (probably temperate deciduous trees) is unhealthy
at the beginning of the historical.

https://webext.cgd.ucar.edu/I20TR/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_hist/lnd/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_hist.1995_2014-clm51_lmbirchwkattgen01ctsm51d006_2deg_GSWP3V1_nocrop_PPE2_hist.1995_2014/set6/set6_crbStock_Eastern_US.png

This suggests to me that the decreasing area of live trees over the
historical is actually probably reflecting trees dying that were already on
verge of dying at end of spinup.  So, I think that this result doesn't go
against our previous hypothesis that with BHS on,  a couple of PFTs are
showing enough of an increase in N limitation associated with cooler soil
temperatures that these plants are not healthy or not surviving.

For reference, here are the full set of diagnostics:
 https://webext.cgd.ucar.edu/I20TR/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_hist/lnd/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_hist.1995_2014-clm51_lmbirchwkattgen01ctsm51d006_2deg_GSWP3V1_nocrop_PPE2_hist.1995_2014/setsIndex.html
<https://webext.cgd.ucar.edu/I20TR/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_hist/lnd/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_hist.1995_2014-clm51_lmbirchwkattgen01ctsm51d006_2deg_GSWP3V1_nocrop_PPE2_hist.1995_2014/setsIndex.html>

**Comment by dlawrenncar** [comment]
Talked with Keith and looked at postAD spinup.  Vegetation carbon drops
rapidly at beginning of AD spinup.  Looks like the AD to postAD worked
fine, but then TOTVEGC just drops everywhere over the course of about 40
years, ending up at a value that is 10% or so of what it should be.
Strangely, LAI, GPP, NPP all look fairly reasonable, so hard to understand
where this is coming from.  Keith is continuing spinup simulations with BHS
off, with BHS off and reverting NSTEM and related params back to default.
With this set of spinups, should be able to establish if the issue is
arising with BHS, with new parameters, or with this tag, which does have
some other changes in it relative to the last tag that worked normally.

**Comment by dlawrenncar** [comment]
Looking a bit further, it looks like there is possibly a crash in mineral
nitrogen in the first few years of postAD.  Been looking at Amazon where
signal is very easy to see.  NPP does drop and allocation drops, so I think
maybe this is carbon starvation.  Strange that GPP, LAI recover to normal
levels by year 30-40.  New runs will be instructive, hopefully.

**Comment by olyson** [comment]
Diagnostics for the first 100 years of postAD of PPE compared to lmbirch postAD:

https://webext.cgd.ucar.edu/I1850/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_1850pAD/lnd/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_1850pAD.471_500-clm51_lmbirchwkattgen01ctsm51d006_2deg_GSWP3V1_nocrop_PPE_1850pAD.271_300/setsIndex.html

**Comment by wwieder** [comment]
It's hard to tell what ecosystem responses are leading vs. following, but
Amazon also has big changes in 2m air temperature righ
<https://webext.cgd.ucar.edu/I1850/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_1850pAD/lnd/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_1850pAD.471_500-clm51_lmbirchwkattgen01ctsm51d006_2deg_GSWP3V1_nocrop_PPE_1850pAD.271_300/set6/set6_landf_Amazonia.png>t
off the bat in the 'new' simulation.  It would be interesting to see if
this occurs in the BHSoff simulation too.

You don't see the temperature changes in NE US, but N costs
<https://webext.cgd.ucar.edu/I1850/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_1850pAD/lnd/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_1850pAD.471_500-clm51_lmbirchwkattgen01ctsm51d006_2deg_GSWP3V1_nocrop_PPE_1850pAD.271_300/set6/set6_cnFlx_Eastern_US.png>seem
to spike in both regions
<https://webext.cgd.ucar.edu/I1850/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_1850pAD/lnd/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE_1850pAD.471_500-clm51_lmbirchwkattgen01ctsm51d006_2deg_GSWP3V1_nocrop_PPE_1850pAD.271_300/set6/set6_cnFlx_Amazonia.png>
with the 'new' simulation (see lower right panel).

**Comment by olyson** [comment]
I've put spinup plots (AD and pAD) for these recent simulations on the bottom of the landing page:

https://www.cgd.ucar.edu/tss/clm/diagnostics/clm5.1_dev/index.html

The BHSOff (BHS turned off) simulation has the same problem as the leafbiomassesai simulation, where TOTVEGC drops rapidly in the first few decades of the pAD spinup.
So this appears to be a problem with the tag rather than with BHS.
I have another pAD simulation in progress with BHSOff but with the original nstem values.

**Comment by olyson** [comment]
BHSOff with original stem values has the same problem.

**Comment by dlawrenncar** [comment]
Thanks.  In this case, I am wondering if this might be related to this bug
fox for  wood c:n ratios.  The fact that this seems to be affecting just
deadstemc (I think) and then propagating through a few other things makes
me suspicious.

https://github.com/ESCOMP/CTSM/issues/443

Unfortunately, I'm not well versed enough in github to sort out quickly
when this bug was integrated into the code (I'd like someone to teach me!),
but I think it is possible that it wasn't included in the so-called LMBIRCH
simulations.

I will say, though, that looking through that issue, it does look like Will
did some test simulations with the fix and things looked fine.  But I don't
know if those tests were done with the actual tag or with code mods.  Seems
possible that there is a bug in the bug fix (?).  Or, that there is
something particular about coming out of AD spinup mode that intersects
with this bug fix that was never really tested/evaluated.

**Comment by olyson** [comment]
It looks like that bug fix went into ctsm1.0.dev109.  The lmbirch simulation used  ctsm5.1.dev006 and the PPE simulations are using ctsm5.1.dev021.  I checked the code itself and it looks like the changes are in both those code bases.

I ran a pAD with BHS off with ctsm5.1.dev021 (the base tag for the PPE tag), initialized from the BHSOffnstemorig run, and TOTVEGC was fine.
So seems like we should be able to figure this out by looking at the differences between PPE.n02_ctsm5.1.dev021 and ctsm5.1.dev021 code, and the parameter files.

**Comment by dlawrenncar** [comment]
Ok. Not sure I understand for sure, but I think you are saying that it’s
not the wood cn bug, then.

On Feb 6, 2021, at 3:35 PM, Keith Oleson <notifications@github.com> wrote:

﻿

It looks like that bug fix went into ctsm1.0.dev109. The lmbirch simulation
used ctsm5.1.dev006 and the PPE simulations are using ctsm5.1.dev021. I
checked the code itself and it looks like the changes are in both those
code bases.

I ran a pAD with BHS off with ctsm5.1.dev021 (the base tag for the PPE
tag), initialized from the BHSOffnstemorig run, and TOTVEGC was fine.
So seems like we should be able to figure this out by looking at the
differences between PPE.n02_ctsm5.1.dev021 and ctsm5.1.dev021 code, and the
parameter files.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ESCOMP/CTSM/issues/1274#issuecomment-774553719>, or
unsubscribe

**Comment by olyson** [comment]
Right.

**Comment by wwieder** [comment]
I'll start with a quick solution to this problem, then explain more:

**two parameters  `rf_cwdl2_bgc` = 0  and `rf_cwdl3_bgc` = 0**
**Changing both of these to 0.5 may help?**

I've been thinking a bit about the woodCN hypothesis.  These changes should make much of an impact on the the veg side (which should actually be less N limited with the changes in the woodCN bug fix).  Instead, they would potentially impact the  decay of coarse woody debris (CWD), which now has a much higher C:N ratio (500, as parameterized instead of 200-250, which was happening previously).  This would cause a big immobilization of N as CWD moves into litter pools that would be much higher coming out of AD mode (when CWD pools are 10x larger).  The colder soil temperatures in BHS could exacerbate this issue, by reducing N availability, which would be consistent with the feature of the PPE tag of a big crash in mineral AN stocks.

One way to get around this would be to introduce a CO2 flux when CWD  is transferred to litter pools.  Right now we're supposedly representing the physical fragmentation of wood into litter, but this transfer has no associated CO2 flux (which seems pretty absurd).  Putting 50% of this C back to the atmosphere would greatly reduce our N demand associated with decomposition.

**Comment by wwieder** [comment]
Now that I've tried piecing more of the conversation above together it sound like woodCN may not be the culprit?  

One this I don't understand is why it looks like decid trees are OK in 1850, but [dying in the historical period](https://www.cgd.ucar.edu/tss/clm/diagnostics/clm5.1_dev/pft_level_plots/plot_checkfordead_timeseries_PPEFINAL_hist_TLAI.png)?  Something else seems to be going on for C4 grasses, however?   Maybe we need to talk on monday?

**Comment by swensosc** [comment]
I have two single point pAD simulations set up (I hope correctly).  One for
the eastern US, one for amazonia.  The eastern US is mainly pft 7, the
amazonia has a small amount of pft 14 (none of the gridcells in the amazon
have a large percentage of pft 14).  My eastern US died due to wallclock,
so I'm rerunning, but here are plots for amazonia ( focusing on pft 14).
The first is 400 years, the second zooms in on the first 10.
[image: amazonia_full.png]
[image: amazonia_10yr.png]

**Comment by wwieder** [comment]
Hi @swensosc look like these didn't load correctly into github.  maybe just email them?

**Comment by dlawrenncar** [comment]
@wwieder  Just to reiterate, we have determined that nothing is right in 1850.  Vegetation biomass is order magnitude low for all PFTs.  The problem arises after AD spinup over the first 40 years of post AD spinup.  And Keith has isolated to differences between two tags, both of which I believe have the Wood CN bug in it, so it is something in those tags in between.  Need to start exploring the differences.

**Comment by olyson** [comment]
Ok, I've been comparing code.  I see one thing that is incorrect in the PPE.n02_ctsm5.1.dev021.  In CNFireLi2021Mod:

           rootc_col(c) = rootc_col(c) + (frootc(p) + frootc_storage(p) + &
                   frootc_xfer(p) + deadcrootc(p) * 10._r8 +       &
                   deadcrootc_storage(p) + deadcrootc_xfer(p) +    &
                   livecrootc(p)+livecrootc_storage(p) +           &
                   livecrootc_xfer(p))*patch%wtcol(p)

This should be (or at least it was in previous tags):

           if (spinup_state == 2) then
              rootc_col(c) = rootc_col(c) + (frootc(p) + frootc_storage(p) + &
                   frootc_xfer(p) + deadcrootc(p) * 10._r8 +       &
                   deadcrootc_storage(p) + deadcrootc_xfer(p) +    &
                   livecrootc(p)+livecrootc_storage(p) +           &
                   livecrootc_xfer(p))*patch%wtcol(p)
           else
              rootc_col(c) = rootc_col(c) + (frootc(p) + frootc_storage(p) + &
                   frootc_xfer(p) + deadcrootc(p) +                &
                   deadcrootc_storage(p) + deadcrootc_xfer(p) +    &
                   livecrootc(p)+livecrootc_storage(p) +           &
                   livecrootc_xfer(p))*patch%wtcol(p)
           endif

The ctsm5.1.dev021 tag has:

           rootc_col(c) = rootc_col(c) + (frootc(p) + frootc_storage(p) + &
                   frootc_xfer(p) + deadcrootc(p) * spinup_factor_deadwood +       &
                   deadcrootc_storage(p) + deadcrootc_xfer(p) +    &
                   livecrootc(p)+livecrootc_storage(p) +           &
                   livecrootc_xfer(p))*patch%wtcol(p)

Presumably, spinup_factor_deadwood is being set to 1.0 in pAD mode, which might explain why the ctsm5.1.dev021 run seems to behave correctly.

I'm running a test now with the corrected code.  Hopefully that's the only problem.

**Comment by olyson** [comment]
That changed answers, but didn't fix the decline in TOTVEGC.

**Comment by olyson** [comment]
There are similar problems with CNGapMortalityMod and CNVegStrucUpdate.  I've fixed these and the results are looking promising (no decline in TOTVEGC).  I'll continue and post a new pAD spinup plot.

**Comment by dlawrenncar** [comment]
Good work Keith.  Perhaps these errors came in during the aborted attempt
to change the factor of 10 to a constant and the subsequent reversion
wasn't complete.   Or, maybe they came in incorrectly with the updated fire
code.  Once this is sorted out, probably need to think through how to catch
this type of error ... if it's possible.

**Comment by olyson** [comment]
I've posted a plot of the new spinup, looks pretty normal:

https://www.cgd.ucar.edu/tss/clm/diagnostics/clm5.1_dev/spinup_plots/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE3_1850pAD_Spinup-0.png

and survivability timeseries for the first 200 years:

https://www.cgd.ucar.edu/tss/clm/diagnostics/clm5.1_dev/pft_level_plots/plot_checkfordead_timeseries_PPEFINAL3_1850pAD_TLAI.png

**Comment by olyson** [comment]
3.5% disequilibrium:

https://www.cgd.ucar.edu/tss/clm/diagnostics/clm5.1_dev/spinup_plots/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE3_1850pAD_Spinup-0.png

**Comment by olyson** [comment]
Started the historical this morning.

**Comment by ekluzek** [comment]
@olyson latest case: clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE3_hist has changes to the following files:

	modified:   biogeochem/CNFireLi2021Mod.F90
	modified:   biogeochem/CNGapMortalityMod.F90
	modified:   biogeochem/CNVegStructUpdateMod.F90
	modified:   biogeophys/CanopyFluxesMod.F90

The changes in CanopyFluxes are just to remove the iWUE changes. In CNVegStructUpdateMod.F90 they also include setting biomass to the hex 0_r8 rather than double precision 0._r8. We were concerned about what it was doing before.

So to make sure I don't change answers I'll replicate one of @olyson cases and bring in his changes and then make sure anything I do on top of them don't change answers for the base case. I shouldn't have to backout the iWUE changes for example, but better show that it doesn't change the case. In the past I just do a one month test for this sort of thing, which I think should be sufficient here as well.

**Comment by olyson** [comment]
FYI, the code I'm using is PPE.n02_ctsm5.1.021.  We didn't remove the iWUE changes, they just weren't in there yet.  And the 0_r8 was in the code previously, we just moved that section of code up out of the woody if statement.  I agree that it should be 0._r8, which I think you've already done in the latest tag.  Also, note that CNVegSTructUpdateMod also contains mods to htop for spinup.

**Comment by olyson** [comment]
The new historical (clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE3_hist) is done.
The diagnostics landing page is here:

https://www.cgd.ucar.edu/tss/clm/diagnostics/clm5.1_dev/index.html

Standard diagnostics of the new run compared to the lmbirch tag are here:

https://webext.cgd.ucar.edu/I20TR/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE3_hist/lnd/clm51_PPEn02ctsm51d021_2deg_GSWP3V1_leafbiomassesai_PPE3_hist.1995_2014-clm51_lmbirchwkattgen01ctsm51d006_2deg_GSWP3V1_nocrop_PPE2_hist.1995_2014/setsIndex.html

BL deciduous trees and C4 grasses appear to be mostly healthy at present day, although BL deciduous tree survivability did decline a bit during the historical:

https://www.cgd.ucar.edu/tss/clm/diagnostics/clm5.1_dev/pft_level_plots/plot_checkfordead_timeseries_PPEFINAL3_hist_TLAI.png

ILAMB results are in progress.

Here's a table of survivability at the end of the historical for the new run (first row) and lmbirch (last row):

PPEFINAL3  97.104  93.279  98.523  98.485  99.186  94.735  75.171  87.047  96.990  64.218  88.092  72.255  84.214  68.316
LMBIRCH     96.969  92.945  98.254  98.014  99.181  94.628  78.761  87.878  99.578  63.254  90.782  78.014  83.333  73.441

**Comment by dlawrenncar** [comment]
Thanks.  I've been looking through plots and I don't see anything that is surprising.  It does look like 2m air temperature is somewhat cooler across a lot of the globe, especially the high latitudes.  As far as I am concerned, though, everything looks fine and this is ready to be passed over to Forrest and Nate for a new cluster analysis.

**Comment by wwieder** [comment]
i'm looking forward to seeing the ilamb results.  I'm surprised how
productive these new simulations are, I think global GPP is 15-20% higher
now than in CLM5.

Are the PHS parameter changes likely responsible for the decreased latent
heat flux?
And now the model is colder... is this just  because of BHS?

**Comment by olyson** [comment]
ILAMB results are now posted here:

https://webext.cgd.ucar.edu/I20TR/clm50_ctsm10d089_2deg_GSWP3V1_luna3_hist/lnd/_build_CTSM5.1G/index.html

**Comment by olyson** [comment]
Erik, I think the changes I made to CNFireLi2021Mod would also need to be applied to CNFireLi2016Mod and CNFireLi2014Mod, for other versions of the model, right?

**Comment by ekluzek** [comment]
Yes, that's likely correct. I'll make sure that's handled correctly in the final version. It won't matter if you only using clm5_1 for the PPE branch right now. I'm working on making sure the PPE branch matches your simulation results. So I'll be able to make sure this is all straightened out in the end version.

**Comment by dlawrenncar** [comment]
LAI is now more biased, but that was pretty much true with the
LMBIRCH simulation too.  GPP is higher, but from global perspective is
actually closer to the GBAF/FLUXNETMTE product.  Don't think should place
too much weight on that one way or other.  Still looks good to me.

**Comment by wwieder** [comment]
Yeah, was it the LUNA bug fixes that increased LAI globally, or did another
change come in made this shift from CLM5.0?
